src/Apps/CollectorProfile/Routes/Saves2/CollectorProfileSaves2Route.tsx

```diff
@@ -54,6 +55,26 @@ const CollectorProfileSaves2Route: FC<CollectorProfileSaves2RouteProps> = ({
   // Always display "All Saves" artwork list first in the list
   const artworkLists = [allSavesArtworkList, ...customArtworkLists]
 
+  /**
+   * When the artwork list has been successfully deleted,
+   * We also clear all info about the deleted artwork list from the relay store in updater.
+   * For this reason, 404 error will be displayed immediately.
+   * To prevent this from happening, `useMemo` is used here
+   */
+  const selectedArtworkList = useMemo(
+    () => {
+      return artworkLists.find(
+        artworkList => artworkList.internalID === selectedArtworkListId
+      )
+    },
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+    [selectedArtworkListId]
+  )
+
+  if (!selectedArtworkList) {
+    throw new HttpError(404)
+  }
+
```

Comment:
When the artwork list has been successfully deleted, we also clear all info about the deleted artwork list from the relay store in [updater](https://github.com/artsy/force/blob/51fc7478ab56131f47c1c2245e9f89c4abf41c59/src/Apps/CollectorProfile/Routes/Saves2/Components/Actions/Mutations/useDeleteArtworkList.ts#L62) before [redirecting](https://github.com/artsy/force/blob/51fc7478ab56131f47c1c2245e9f89c4abf41c59/src/Apps/CollectorProfile/Routes/Saves2/Components/Actions/DeleteArtworkListModal.tsx#L63) to "All Saves".

For this reason, 404 error will be displayed immediately (see attached video). To prevent this from happening, `useMemo` is used here


https://user-images.githubusercontent.com/3513494/230107210-cb2fcb22-f32e-4152-a062-29ed780a7623.mp4

Reply:
Thanks for explaining 🙏🏽